### PR TITLE
Widget theme implementation and integration

### LINKED ISSUE #20: Widget Theme support
A company should be able to change the theme so that colors in the app reflect the brand colors for that company.

### PR DESCRIPTION
**Pull request by vova-beloded**
Fixes #20 

## Checklist

- [x] Documented
- [x] Unit tested
- [x] Component tested

### CODE REVIEW

File: metrics/test/features/dashboard/presentation/page/dashboard_page_test.dart [hunk 1]
```diff
@@ -62,15 +75,33 @@ void main() {
         );
       },
     );
+
+    testWidgets(
+      "Displays the drawer on tap on menu button",
```
Comment (illia-romanenko) [inline metrics/test/features/dashboard/presentation/page/dashboard_page_test.dart]: Should we add test for theme change? After it's changed - colors are changed as well?

File: metrics/lib/features/common/presentation/metrics_theme/model/metrics_theme_data.dart [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  ///
+  /// [spakrlineTheme] is the color theme of the [SparklineGraph].
+  /// Used to set the default colors to the [SparklineGraph].
+  ///
+  /// [buildResultThemeData] is the color theme for the [BuildResultBarGraph].
+  /// Used to set the colors of the graph bars.
+  const MetricsThemeData({
+    MetricWidgetThemeData circlePercentagePrimaryTheme,
+    MetricWidgetThemeData circlePercentageAccentTheme,
+    MetricWidgetThemeData spakrlineTheme,
```
Comment (illia-romanenko) [inline metrics/lib/features/common/presentation/metrics_theme/model/metrics_theme_data.dart]:
```suggestion
    MetricWidgetThemeData sparklineTheme,
```

File: metrics/lib/features/common/presentation/metrics_theme/model/build_results_theme_data.dart [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const BuildResultsThemeData.light()
+      : canceledColor = Colors.grey,
+        successfulColor = Colors.teal,
+        failedColor = Colors.redAccent;
+
+  /// Creates the default dark color theme of the [BuildResultBarGraph]
+  const BuildResultsThemeData.dark()
+      : canceledColor = Colors.orange,
+        successfulColor = Colors.teal,
+        failedColor = Colors.red;
```
Comment (illia-romanenko) [inline metrics/lib/features/common/presentation/metrics_theme/model/build_results_theme_data.dart]: Suggestion: move this to ThemeData

File: metrics/lib/features/common/presentation/metrics_theme/widgets/metrics_app.dart [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+import 'package:metrics/features/common/presentation/metrics_theme/model/metrics_theme_data.dart';
+import 'package:metrics/features/common/presentation/metrics_theme/widgets/metrics_theme.dart';
+
+enum MetricsThemeType { light, dark }
+
+/// The widget to manage the current [MetricsThemeData].
+class MetricsApp extends StatelessWidget {
```
Comment (illia-romanenko) [inline metrics/lib/features/common/presentation/metrics_theme/widgets/metrics_app.dart]: Suggestion: let's start a discussion about the functionality of this Widget and possibility of merging it with MetricsThemeBuilder

File: metrics/test/features/common/presentation/metrics_theme/model/metrics_theme_data_test.dart [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      expect(copiedTheme.circlePercentagePrimaryTheme, circlePercentageTheme);
+      expect(copiedTheme.circlePercentageAccentTheme, circlePercentageTheme);
+      expect(copiedTheme.sparklineTheme, sparklineTheme);
+      expect(copiedTheme.buildResultTheme, buildResultsTheme);
+      expect(copiedTheme.barGraphBackgroundColor, backgroundColor);
+    },
+  );
+
+  test(
+    "Creates new instance with the same filed if copyWith was used widhout any params",
```
Comment (bootstraponline) [inline metrics/test/features/common/presentation/metrics_theme/model/metrics_theme_data_test.dart]: `widhout` should be without

**Pull request merged**